packages/cdk-global-endpoints/lib/api-domain-name.ts

```diff
@@ -0,0 +1,36 @@
+import * as cdk from 'aws-cdk-lib';
+import * as apigw from 'aws-cdk-lib/aws-apigateway';
+import * as acm from 'aws-cdk-lib/aws-certificatemanager';
+import { Construct } from 'constructs';
+
+export interface ApiGatewayProps {
+    companyDomainName: string;
+    domainName: string;
+    project: string;
+    baseEnv: string;
+    dynamicEnv?: string;
+    certificateArn: string;
+    restApiId: string;
+    basePath?: string;
+}
+
+export class ApiGateway extends Construct {
+    constructor(scope: Construct, id: string, props: ApiGatewayProps) {
+        super(scope, id);
+
+        let customDomain: apigw.DomainName;
+        let cloudformationBasePathMapping: apigw.CfnBasePathMapping;
+
+        customDomain = new apigw.DomainName(this, 'Custom Domain', {
+            domainName: `${props.domainName}`,
+            certificate: acm.Certificate.fromCertificateArn(this, "Certificate", `${props.certificateArn}`),
+            endpointType: apigw.EndpointType.REGIONAL,
+            securityPolicy: apigw.SecurityPolicy.TLS_1_2
+        })
+
+        cloudformationBasePathMapping = new apigw.CfnBasePathMapping(this, "CloudformationBasePathMapping", {
```

Comment:
This custom domain won't actually be applied to an API. Also if we can avoid using constructs prefixed `Cfn` that would be preferable. These are just straight cloudformation objects. We would prefer to use the L1 constructs wherever possible.